### ISSUE TITLE
Could your module fix microphone gain dropping?

### ISSUE THREAD

**Issue by ghost** [reiryuki/Mic-Fix-Magisk-Module#1]
Hello.
I've bought a new lavalier microphone to use with my Redmi Note 4 (mido), specifically Boya BY-M1 Pro but when I speak/sing loudly (around -10 dB), recording volume (gain) drops to -20dB and after a few seconds it rises again back at the whatever volume level I'm speaking/singing.
I am sure that the microphone isn't at fault since it works as expected when connected to my computer or my iPhone. I also tried switching custom ROM but it didn't fix it.
I suppose that this bug has to do with somekind of auto-gain control.

Could you program your Mic-Fix-Magisk-Module to also fix this issue, perhaps by disabling the AutomaticGainControl?

**Comment by reiryuki** [comment]
Have you tried this module?
Send your \*mixer\*path\*.xml files

**Comment by ghost** [comment]
Yes, I have already tried your module but it didn't fix it. To test it, I plug in my microphone, go into the recording app and I make a noise at a steady tone. Initially, sound level is at -10dB and after a second, it suddenly drops to -20dB. I was hoping your module could fix that but it didn't work. Where can I find my *mixer*path*.xml files?
I think the issue is with automatic gain control (AGC) in my phone. My microphone works perfectly on my PC and other phones. The only other possibility is that it is a hardware issue with my Redmi Note 4.

**Comment by reiryuki** [comment]
/vendor/etc/

where you find AGC? Screenshot

**Comment by ghost** [comment]
Sorry but the only folder I could find was /vendor/lib64, there aren't any xml files in /vendor or /vendor/lib64. My device is Redmi Note 4 (mido) running LineageOS 18.1 UNOFFICIAL.
I suspect that the AGC is at fault because I read that:

> Automatic Gain Control (AGC) is an audio pre-processing which automatically normalizes the output of the captured signal by boosting or lowering input from the microphone to match a preset level so that the output signal level is virtually constant.

**Comment by reiryuki** [comment]
/vendor/etc/
not /vendor/ or /vendor/lib*/

I already deactivates agc pre process stream in my soundfx modules

**Comment by ghost** [comment]
> /vendor/etc/
> not /vendor/ or /vendor/lib*/

Let me rephrase it, I can't find a /vendor/etc/ folder (I tried finding it in OrangeFox Recovery), it seems that it doesn't exist.

> I already deactivates agc pre process stream in my soundfx modules

So I should rule out AGC being at fault? Then perhaps it's a hardware issue. I also tried chaning custom ROM, going back to older Android but that didn't fix it. Only other solution would be to return to stock ROM but I wouldn't want to.
Anyway, thanks for your help, I suppose it's not solvable.

**Comment by reiryuki** [comment]
Use root explorer not recovery to find. Send audio_effects****** files also.

I only deactivate agc at pre process stream, not completely deactivated. I will send you a test module later

**Comment by ghost** [comment]
[files.zip](https://github.com/reiryuki/Mic-Fix-Magisk-Module/files/6479684/files.zip)

**Comment by reiryuki** [comment]
No agc there. You may have other audio_effects files.
Find at /system/etc/ also.

**Comment by ghost** [comment]
The only file I could find related to audio effects:
[audio_effects.zip](https://github.com/reiryuki/Mic-Fix-Magisk-Module/files/6483318/audio_effects.zip)

**Comment by reiryuki** [comment]
The rom/vendor doesn't use agc

**Comment by ghost** [comment]
So does that mean that i probably is a hardware issue with Redmi Note 4? Thanks for your help and sorry to have put you into all this trouble.

**Comment by reiryuki** [comment]
I don't know actually
You can send your logcats
